scalafix-core/shared/src/main/scala/scalafix/internal/rule/RuleCtxImpl.scala

```diff
@@ -48,18 +51,28 @@ case class RuleCtxImpl(tree: Tree, config: ScalafixConfig) extends RuleCtx {
     logger.elem(values: _*)
   }
 
-  def printLintMessage(msg: LintMessage, owner: RuleName): Unit = {
+  def reportLintMessage(msg: LintMessage, owner: RuleName): Boolean = {
```

Comment:
Maybe we should just collect the message to be reported here (eg returning (Message, Positon, Severity, Escaped | Not Escaped) )